src/groups/mqb/mqbnet/mqbnet_tcpsessionfactory.cpp

```diff
@@ -241,6 +242,95 @@ void stopChannelFactory(mwcio::ChannelFactory* channelFactory)
     factory->stop();
 }
 
+class TcpInterfaceConfigValidator {
+  public:
+    // TYPES
+    typedef bsl::allocator<unsigned char> allocator_type;
+
+  private:
+    // PRIVATE TYPES
+    class UniqueProperty {
+      public:
+        typedef bsl::allocator<unsigned char> allocator_type;
+
+      private:
+        allocator_type d_allocator;
+
+      public:
+        UniqueProperty(allocator_type allocator = allocator_type())
+        : d_allocator(allocator)
+        {
+        }
+
+        allocator_type get_allocator() const { return d_allocator; }
+
+        /// viewFn :: NetworkInterface -> T
+        template <typename It, typename F>
+        bool operator()(It begin, It end, F viewFn) const
+        {
+            typedef typename bsl::iterator_traits<It>::reference    Reference;
+            typedef typename bsl::invoke_result<F, Reference>::type Result;
+            bsl::vector<Result> sorted(d_allocator);
+            sorted.reserve(bsl::distance(begin, end));
+
+            bsl::transform(begin, end, bsl::back_inserter(sorted), viewFn);
+            bsl::sort(sorted.begin(), sorted.end());
+            typename bsl::vector<Result>::iterator newEnd =
+                bsl::unique(sorted.begin(), sorted.end());
+            return sorted.end() == newEnd;
+        }
+    };
+
+    // PRIVATE DATA
+    UniqueProperty d_unique;
+
+    // PRIVATE STATIC FUNCTIONS
+    static int portView(const mqbcfg::TcpInterfaceListener& listener)
```

Comment:
It could create a view, so technically no. The codegen'd API doesn't give me const ref access to this field, so it's a little unfortunate, but It's sufficient enough for our purposes.